R/apply_fmt_fn.R

```diff
@@ -55,15 +57,17 @@ apply_fmt_fn <- function(x) {
 #' @param x (`integer`, `string`, or `function`)\cr
 #'   a non-negative integer, string alias, or function
 #' @param call (`environment`)\cr
-#'   frame for error messaging. Default is [parent.frame()].
+#'   frame for error messaging. Default is [get_cli_abort_call()].
 #'
 #' @return a function
 #' @export
 #'
 #' @examples
 #' alias_as_fmt_fn(1)
 #' alias_as_fmt_fn("xx.x")
-alias_as_fmt_fn <- function(x, call = parent.frame()) {
+alias_as_fmt_fn <- function(x, call = get_cli_abort_call()) {
```

Comment:
🤔 I think the primary reason we needed the `call` argument here was to pass along the calling frame from one function to another. With this functionality you wrote, I think we can just remove the call arg altogether? What do you think?

Reply:
I think it makes sense to remove the `call` argument where possible - I went ahead and did that. Note that there are also a few instances of parameter `env = caller_env()` which I didn't touch since I think they are necessary and break if changed.